packages/trader/src/AppV2/Containers/ContractDetails/__tests__/contract-details.spec.tsx

```diff
@@ -0,0 +1,257 @@
+    OrderDetails.displayName = 'OrderDetails';
+    return OrderDetails;
+});
+
+jest.mock('@deriv/shared', () => ({
+    isValidToSell: jest.fn(),
+    isMultiplierContract: jest.fn(),
+    isValidToCancel: jest.fn(),
+    isOpen: jest.fn(),
+    useWS: jest.fn(),
+    hasContractEntered: jest.fn(),
+    isForwardStarting: jest.fn(),
+    isAccumulatorContract: jest.fn(),
+    WS: {
+        contractUpdateHistory: jest.fn(),
+    },
+}));
+jest.mock('AppV2/Utils/contract-details-config', () => ({ getContractDetailsConfig: jest.fn() }));
+
+const mockContractInfo = {
+    contract_id: 1,
+    currency: 'USD',
+    contract_type: 'multiplier',
+    limit_order: {},
+    purchase_time: Date.now(),
+    shortcode: 'mock_shortcode',
+};
+
+const default_mock_store = {
+    modules: {
+        trade: {
+            active_symbols: [{ symbol: 'R_10' }],
+        },
+    },
+    ui: {
+        addToast: jest.fn(),
+        current_focus: '',
+        is_mobile: true,
+        removeToast: jest.fn(),
+        setCurrentFocus: jest.fn(),
+        should_show_cancellation_warning: false,
+        toggleCancellationWarning: jest.fn(),
+    },
+    common: {
+        server_time: moment('2023-11-21 10:59:59'),
+    },
+    contract_trade: {
+        getContractById: jest.fn(),
+    },
+};
+
+describe('ContractDetails', () => {
+    beforeEach(() => {
+        (useContractDetails as jest.Mock).mockReturnValue({
+            contract_info: mockContractInfo,
+            is_loading: false,
+        });
+        (useOrderDetails as jest.Mock).mockReturnValue({
+            details: {},
+        });
+        (isValidToSell as jest.Mock).mockReturnValue(true);
+        (isMultiplierContract as jest.Mock).mockReturnValue(true);
+        (isValidToCancel as jest.Mock).mockReturnValue(true);
+        (isOpen as jest.Mock).mockReturnValue(true);
+        (hasContractEntered as jest.Mock).mockReturnValue(true);
+        (isForwardStarting as jest.Mock).mockReturnValue(true);
+        (isAccumulatorContract as jest.Mock).mockReturnValue(false);
+        (useWS as jest.Mock).mockReturnValue({
+            send: jest.fn(),
+            subscribe: jest.fn(),
+            unsubscribe: jest.fn(),
+        });
+        (WS.contractUpdateHistory as jest.Mock).mockResolvedValue({
+            contract_update_history: [],
+        });
+        (getContractDetailsConfig as jest.Mock).mockReturnValue({
+            is_take_profit_visible: true,
+            is_stop_loss_visible: true,
+            is_deal_cancellation_visible: true,
+            is_tp_history_visible: true,
+        });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const renderContractDetails = () => {
+        render(
+            <TraderProviders store={mockStore(default_mock_store)}>
+                <ContractDetails />
+            </TraderProviders>
+        );
+    };
+
+    it('should render the ContractCard component', async () => {
+        await act(async () => {
```

Comment:
Will await waitFor(...) work?

Reply:
Same in other places

Reply:
Again, what is the point of this change? What is the benefit? Programming can be done in more than one way and it seems we have both of these ways used in the system already. I'm closing this unless there's an explanation on why.